### PR TITLE
Share code in completion tests.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide

File: src/EditorFeatures/Test/Completion/AbstractCompletionProviderTests.cs [hunk 1]
```diff
@@ -58,6 +58,12 @@ internal CompletionServiceWithProviders GetCompletionService(Workspace workspace
         internal abstract CompletionServiceWithProviders CreateCompletionService(
             Workspace workspace, ImmutableArray<CompletionProvider> exclusiveProviders);
 
+        protected abstract string ItemPartiallyWritten(string expectedItemOrNull);
+
+        protected abstract Task<TestWorkspace> CreateWorkspaceAsync(string fileContents);
+
+        protected abstract Task BaseVerifyWorkerAsync(string code, int position, string expectedItemOrNull, string expectedDescriptionOrNull, SourceCodeKind sourceCodeKind, bool usePreviousCharAsTrigger, bool checkForAbsence, int? glyph);
```
Comment (dpoeschl) [inline src/EditorFeatures/Test/Completion/AbstractCompletionProviderTests.cs]: The "Base" prefix feels weird on abstract methods.

Comment (CyrusNajmabadi) [inline src/EditorFeatures/Test/Completion/AbstractCompletionProviderTests.cs]: i don't like how this test class works *at all*.  It has baked in logic, then has classes that derive from that and specialize, and then there are even deeper subclasses that then do *not* want that specialization.  It's basically 100% broken.

**Comment by CyrusNajmabadi** [comment]
@dotnet-bot retest windows_eta_open_prtest please
// Previous failure: http://dotnet-ci.cloudapp.net/job/Private/job/dotnet_roslyn-internal/job/master/job/windows_eta_open_prtest/356/
// Retest reason: error MSB4018: The "CreateZipPackage" task failed unexpectedly.

**Comment by rchande** [comment]
Can you please hold off on this until the target typing change is in?

**Comment by rchande** [comment]
(There are going to be a lot of merge conflicts)

**Comment by CyrusNajmabadi** [comment]
test vsi please

**Comment by CyrusNajmabadi** [comment]
test vsi please

**Comment by CyrusNajmabadi** [comment]
@dotnet-bot retest linux_debug_prtest please
// Previous failure: http://dotnet-ci.cloudapp.net/job/dotnet_roslyn/job/master/job/linux_debug_prtest/455/
// Retest reason:

**Comment by CyrusNajmabadi** [comment]
retest windows_debug_unit32_prtest  please

**Comment by CyrusNajmabadi** [comment]
retest windows_debug_unit64_prtest please

**Comment by CyrusNajmabadi** [comment]
@dotnet-bot retest windows_debug_unit32_prtest please
// Previous failure: http://dotnet-ci.cloudapp.net/job/dotnet_roslyn/job/master/job/windows_debug_unit32_prtest/469/
// Retest reason:

**Comment by CyrusNajmabadi** [comment]
@dotnet-bot retest windows_debug_unit64_prtest please
// Previous failure: http://dotnet-ci.cloudapp.net/job/dotnet_roslyn/job/master/job/windows_debug_unit64_prtest/466/
// Retest reason:

**Pull request merged**